### DISCUSSION TITLE
is strongswan 5.7.2 which use VPP for encryption/decryption need --kernel-libipsec plug-in

### DISCUSSION THREAD

**Discussion by antonymsouza** [strongswan/strongswan#2378] [General]
I can see Fake UDP encapsulation on ESP packets. I understand from https://wiki.strongswan.org/issues/3223. it always enforce NAT-T situation if the --kernel-libipsec plug-in enabled. As we are using VPP for the encryption/decryption, do we really need this plug-in. --enable-kernel-vpp --enable-socket-vpp --enable-kernel-libipsec flags enabled in ./configure

**Comment by tobiasbrunner** [comment]
You refer to VPP, which isn't supported by strongSwan (there is an experimental branch, but no released version supports it), so you might be using some out-of-tree patches. Then you refer to the _kernel-libipsec_ plugin, which has absolutely nothing to do with VPP. So if you're asking if you need that plugin, the answer is most likely no.

**Reply by antonymsouza** [reply]
Thanks for the response.
